pkg/state/chart_dependency.go

```diff
@@ -108,28 +111,16 @@ func (d *ResolvedDependencies) add(dep ResolvedChartDependency) error {
 	return nil
 }
 
-func (d *ResolvedDependencies) Get(chart, versionConstraint string) (string, error) {
-	if versionConstraint == "" {
-		versionConstraint = "*"
-	}
-
+func (d *ResolvedDependencies) Get(chart, release string) (string, error) {
 	deps, exists := d.deps[chart]
 	if exists {
 		for _, dep := range deps {
-			constraint, err := semver.NewConstraint(versionConstraint)
-			if err != nil {
-				return "", err
-			}
-			version, err := semver.NewVersion(dep.Version)
-			if err != nil {
-				return "", err
-			}
-			if constraint.Check(version) {
+			if dep.Alias == release {
```

Comment:
@olivierboudet All in all, can we do this? This way, you still have the freedom to omit the version in helmfile.yaml so that the version stored in the lock file is always used.

```suggestion
			if dep.Alias != release {
				continue
			}
			constraint, err := semver.NewConstraint(versionConstraint)
			if err != nil {
				return "", err
			}
			version, err := semver.NewVersion(dep.Version)
			if err != nil {
				return "", err
			}
			if constraint.Check(version) {
```

Reply:
yes I think it is good to do so. I will push a new commit which will reintroduce version check.